src/lib.rs

```diff
@@ -194,10 +194,9 @@
 //! for Async Rust. We have a long journey ahead of us. But we're excited you're here with us!
 
 #![cfg_attr(feature = "docs", feature(doc_cfg))]
-// #![warn(missing_docs)]
-#![warn(missing_debug_implementations, rust_2018_idioms)]
-#![doc(test(attr(deny(rust_2018_idioms, warnings))))]
-#![doc(test(attr(allow(unused_extern_crates, unused_variables))))]
+#![forbid(unsafe_code, rust_2018_idioms)]
+#![deny(missing_debug_implementations, nonstandard_style)]
+#![warn(missing_docs, unreachable_pub, future_incompatible)]
```

Comment:
```suggestion
#![forbid(unsafe_code)]
#![deny(missing_debug_implementations, nonstandard_style)]
#![warn(missing_docs, unreachable_pub, future_incompatible, rust_2018_idioms)]
```

As said in https://github.com/taiki-e/pin-project/issues/281#issuecomment-699543730, I do not recommend using `forbid` on the lints other than `unsafe_code` lint.